pkg/tree/tree.go

```diff
@@ -27,6 +30,29 @@ func (t Tree) String() string {
 	return s
 }
 
+// GetString returns the string value at the given path
+func (t Tree) GetString(path ...string) (string, error) {
+	if len(path) == 1 {
+		// check if exists
+		val, exist := t[path[0]]
+		if exist {
+			// check if string
+			if reflect.TypeOf(val).Kind() == reflect.String {
+				return val.(string), nil
+			}
+			return "", fmt.Errorf("expected string but found a different type")
+		}
+		return "", fmt.Errorf("could not find %s node", path[0])
+	}
+
+	// check if exists
+	val, exist := t[path[0]]
+	if exist {
+		return val.(Tree).GetString(path[1:]...)
```

Comment:
I think you also need to do a typesafe cast to Tree here.  For example, if try to get path ("a", "b", "c") where the value of "a" is a string (leaf) instead of a subtree, this cast will fail.